### PR TITLE
feat: add batch-create-offers page

### LINKED ISSUE #712: Batch create offers from CSV
image

- CSV file example: https://docs.google.com/spreadsheets/d/1g8thQ4Bdt259x5Xrl-6wYEYGs8AdxWtGWgxXVplnEg0/edit#gid=0
- 1st step is implemented by https://github.com/bosonprotocol/offers-csv-parser

### PR DESCRIPTION
**Pull request by levalleux-ludo**
fixes #712 

The CSV file needs to be parsed and prepared (to a JSON file) using https://github.com/bosonprotocol/offers-csv-parser

Then, 
- connect to the dApp as a Seller (note: the seller account needs to exist)
- click the link "Batch Create Products" in the homepage's footer 
  image
- upload the JSON file created above
- click Create Offers (<-- button is disabled in case all the offers are already existing, preventing reusing the same JSON file to create redundant offers)
  image

### CODE REVIEW

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    <div>
+      <label htmlFor="files">Choose files</label>{" "}
+      <input
+        ref={fileRef}
+        multiple={true}
+        type="file"
+        name={name}
+        onChange={onChange}
+      />
+    </div>
```
Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:32]: maybe we can use the existing Upload component?

File: src/components/modal/components/RelistOfferModal/RelistOfferModal.tsx [hunk 2]
```diff
@@ -176,7 +176,6 @@ export const RelistOfferModal: React.FC<RelistOfferModalProps> = ({
           };
 
           await createOffers({
-            isMultiVariant,
```
Comment (albertfolch-redeemeum) [inline src/components/modal/components/RelistOfferModal/RelistOfferModal.tsx:179]: why?

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  );
+
+  const offerMetadatas = useGetOfferMetadata(
+    { metadataUris: offersList.map((offer) => offer.metadataUri) },
+    {
+      enabled: !!offersList && offersList.length > 0
+    }
+  );
+
+  useMemo(async () => {
```
Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:136]:
```suggestion
  useEffect(async () => {
```

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+              />
+            </FormField>
+          </ContainerProductPage>
+          <DetailTable data={tableData}></DetailTable>
+          {invalidFile && <p>Invalid file</p>}
+          <ButtonsSection>
+            <BosonButton
+              type="submit"
+              variant="primaryFill"
+              disabled={!offersToBeCreated || offersToBeCreated.length == 0}
```
Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:262]:
```suggestion
              disabled={!offersToBeCreated || !offersToBeCreated.length}
```

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            >
+              Create Offers
+            </BosonButton>
+          </ButtonsSection>{" "}
```
Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:266]:
```suggestion
          </ButtonsSection>
```

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+          errorMessage: "Something went wrong"
+        });
+      }
+    }
+  };
+
+  return (
+    <>
+      <Formik
+        initialValues={{ files: "" }}
```
Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:236]:
```suggestion
        initialValues={{ files: [] }}
```

Comment (albertfolch-redeemeum) [inline src/pages/batch-create-offers/BatchCreateOffers.tsx:236]: add validationSchema if possible

**Review by albertfolch-redeemeum — APPROVED** [review]

File: src/pages/batch-create-offers/BatchCreateOffers.tsx [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+        setInvalidFile(false);
+      } catch (e) {
+        setOffersList([]);
+        setInvalidFile(true);
+      }
+    } else {
+      setOffersList([]);
+      setInvalidFile(false);
+    }
+  }, [selectedFile]);
```
[the remark this answers is not in the document]
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx:113]: ok

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 3 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx:136]: ok

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 4 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx:262]: ok

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 5 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx:266]: ok

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 6 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx:236]: ok

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 1 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx]: I've tried but I find the Upload component is a bit too much oriented for images
image
Personally I have a preference for keeping the basic input...

Comment (albertfolch-redeemeum) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx]: true, nvm

File: src/pages/batch-create-offers/BatchCreateOffers.tsx (hunk 6 above)
Comment (levalleux-ludo) [reply src/pages/batch-create-offers/BatchCreateOffers.tsx]: Adding a validation schema here breaks the logic I've implemented.
At the moment the Create Offers button is the Form submission command. But managing its status (being enable or disable) is outside of the Form scope (it requires to read the selected file, to parse all offers there, and check if there are some offers to be created - we disable if all offers are already existing).
That's why I'm managing the validity my own way

**Pull request merged**